cranelift/filetests/src/test_rodata.rs

```diff
@@ -64,45 +61,45 @@ impl SubTest for TestRodata {
         );
 
         // Run final code through filecheck.
-        let text = format!("{:X?}", sink.rodata);
+        let text = format!("{:X?}", sink.rodata());
         info!("Found rodata: {}", text);
         run_filecheck(&text, context)
     }
 }
 
 /// Code sink that only captures emitted rodata
+#[derive(Default)]
 struct RodataSink {
-    in_rodata: bool,
-    rodata: Vec<u8>,
+    emitted: Vec<u8>,
+    rodata_offset: Option<usize>,
+}
+
+impl RodataSink {
+    fn rodata(&self) -> Vec<u8> {
+        let rodata_offset = self.rodata_offset.expect("Expected rodata to have a known offset (has the section been started with begin_rodata()?)");
+        self.emitted[rodata_offset..].to_vec()
+    }
 }
 
 impl binemit::CodeSink for RodataSink {
     fn offset(&self) -> binemit::CodeOffset {
-        0
+        self.emitted.len() as u32
```

Comment:
Alternatively, what do you think of having the `RodataSink` struct have one more `usize` field called `offset`? It would be returned by this function, and it would need to be updated by the functions below to take into account the offset sizes. This would have the benefit of storing less bytes in the `emitted` vector, so it'd be more efficient and would make the tests run faster (no need for resizing, for instance).